### ISSUE TITLE
Multiple outputs with --html crashes with "TypeError: Object of type License is not JSON serializable"

### ISSUE THREAD

**Issue by MartinPetkov** [nexB/scancode-toolkit#1199]
To reproduce:

```bash
$ git checkout v2.9.3
$ ./configure
$ source ./bin/activate
(scancode-toolkit) $ cd /tmp/
(scancode-toolkit) $ git clone https://github.com/madler/zlib && cd zlib/
(scancode-toolkit) $ scancode --verbose --license --json-pp ./scancode_results.json --html ./scancode_results.html .
```

This results in:

```bash
...
Save scan results...
 Save scan results as: html...
 Save scan results as: json-pp...
ERROR: failed to run output plugin: json-pp:
Traceback (most recent call last):
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/src/scancode/cli.py", line 809, in run_plugins
    plugin.process_codebase(codebase, **kwargs)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/src/formattedcode/output_json.py", line 99, in process_codebase
    pretty=True)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/src/formattedcode/output_json.py", line 140, in write_json
    output_file.write(simplejson.dumps(scan, **kwargs))
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/__init__.py", line 399, in dumps
    **kw).encode(obj)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 298, in encode
    chunks = list(chunks)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 696, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 708, in _iterencode
    for chunk in _iterencode_list(o, _current_indent_level):
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 531, in _iterencode_list
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 531, in _iterencode_list
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 716, in _iterencode
    o = _default(o)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 273, in default
    o.__class__.__name__)
TypeError: Object of type License is not JSON serializable

Scanning done.
Some files failed to scan properly:
ERROR: failed to run output plugin: json-pp:
Traceback (most recent call last):
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/src/scancode/cli.py", line 809, in run_plugins
    plugin.process_codebase(codebase, **kwargs)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/src/formattedcode/output_json.py", line 99, in process_codebase
    pretty=True)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/src/formattedcode/output_json.py", line 140, in write_json
    output_file.write(simplejson.dumps(scan, **kwargs))
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/__init__.py", line 399, in dumps
    **kw).encode(obj)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 298, in encode
    chunks = list(chunks)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 696, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 708, in _iterencode
    for chunk in _iterencode_list(o, _current_indent_level):
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 531, in _iterencode_list
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 531, in _iterencode_list
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 652, in _iterencode_dict
    for chunk in chunks:
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 716, in _iterencode
    o = _default(o)
  File "/usr/local/google/home/mpetkov/development/scancode-dev/scancode-toolkit/local/lib/python2.7/site-packages/simplejson/encoder.py", line 273, in default
    o.__class__.__name__)
TypeError: Object of type License is not JSON serializable
...
```

It happens with any combination of `--html` and another output type.

It does not happen if only using the `--html` output and nothing else.

It didn't happen on older versions but does appear to happen on v2.9.3 and above (i.e. develop 02c2ffb6033c78e617dcbc2acc89a88926caf458).

**Comment by pombredanne** [comment]
@MartinPetkov thanks ++ for this bug report: great find! Let me check into this

**Comment by MartinPetkov** [comment]
Thanks for the test, I didn't think to add one. Let me know how I can help!

**Comment by pombredanne** [comment]
@MartinPetkov I think the bug is fixed... though a better fix would be needed. At least you can test this branch

**Comment by pombredanne** [comment]
FWIW, the key fix is this innocuous bit https://github.com/nexB/scancode-toolkit/commit/bb362f4764caa03a4d08e3785d2a19246a0b9afd#diff-0d7c76a6af49da6b0cd751a3d6d8d68fR235

**Comment by pombredanne** [comment]
@MartinPetkov this has been merged in develop. Feedback welcomed!

**Comment by pombredanne** [comment]
Cosing as this has also been released since v2.9.4 tag
